src/decoder/stream.rs

```diff
@@ -1227,35 +1136,23 @@ impl StreamingDecoder {
             ColorType::Indexed => {
                 // The transparency chunk must be after the palette chunk and
                 // before the data chunk.
-                if info.palette.is_none() {
-                    return Err(DecodingError::Format(
-                        FormatErrorInner::BeforePlte { kind: chunk::tRNS }.into(),
-                    ));
-                } else if self.have_idat {
-                    return Err(DecodingError::Format(
-                        FormatErrorInner::OutsidePlteIdat { kind: chunk::tRNS }.into(),
-                    ));
+                if info.palette.is_some() || !self.have_idat {
+                    info.trns = Some(Cow::Owned(vec));
                 }
-
-                info.trns = Some(Cow::Owned(vec));
                 Ok(Decoded::Nothing)
             }
-            c => Err(DecodingError::Format(
-                FormatErrorInner::ColorWithBadTrns(c).into(),
-            )),
+            _ => {
+                // Mimicking how `png_handle_tRNS` in `libpng` treats this as a benign error.
+                Ok(Decoded::Nothing)
+            }
         }
     }
 
     fn parse_phys(&mut self) -> Result<Decoded, DecodingError> {
         let info = self.info.as_mut().unwrap();
-        if self.have_idat {
-            Err(DecodingError::Format(
-                FormatErrorInner::AfterIdat { kind: chunk::pHYs }.into(),
-            ))
-        } else if info.pixel_dims.is_some() {
-            Err(DecodingError::Format(
-                FormatErrorInner::DuplicateChunk { kind: chunk::pHYs }.into(),
-            ))
+        if !self.have_idat && info.pixel_dims.is_none() {
```

Comment:
I think the `if` and `else` blocks are swapped here?

Reply:
Ooops.  Thanks for catching this.  Let me fix this change when rebasing on top of the tip-of-tree.

I'll write a note to myself to try to improve unit test coverage for auxiliary chunks.  (Sadly I am not sure if I'll have time for it this week, but I'll keep it in my to-do list.)